src/Analyser/Analyser.php

```diff
@@ -142,6 +142,8 @@ public function analyse(
 			return $errors;
 		}
 
+		$errors = [];
```

Comment:
Isn't this defined in L118

Reply:
It is - not necessary to add the variable again.

Reply:
It absolutely is.

Because it's used to hold an array of strings, however, a static analysis tool that doesn't understand the impact of `if (count($arr) > 0) { return; }` on `$arr` (which is all of them, including PHPStan) will think that `$errors` contains both `string` and `PHPStan\Analyser`.

This isn't an issue in PHPStan currently because it types `$errors` as `mixed[]`. Once https://github.com/phpstan/phpstan/issues/747 is fixed, PHPStan will get confused without this re-initialisation.

An alternative is to change the name of `$errors` so the variable isn't reused for a different purpose – other languages wouldn't allow this.

Reply:
Ticketed here - https://github.com/vimeo/psalm/issues/442. A little hard to repro in PHPStan, but this is effectively it: https://phpstan.org/r/0a652e035b24417d361fb77ad8ba6df3

Reply:
OK, but two variables of different names would be better and less WTF.